### PR TITLE
Add switch

### LINKED ISSUE #29: Switch

### PR DESCRIPTION
**Pull request by Pante**
Closes #29

### CODE REVIEW

File: forui/lib/src/widgets/switch.dart [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+  /// drag behavior feel slightly more reactive.
+  ///
+  /// By default, the drag start behavior is [DragStartBehavior.start].
+  final DragStartBehavior dragStartBehavior;
+
+  /// Creates a [FSwitch].
+  const FSwitch({
+    required this.value,
+    required this.onChanged,
+    super.key,
```
Comment (kawaijoe) [inline forui/lib/src/widgets/switch.dart:82]: Should super be the last element?

File: forui/lib/src/widgets/switch.dart [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+  final FocusNode? focusNode;
+
+  /// Handler called when the focus changes.
+  ///
+  /// Called with true if this widget's node gains focus, and false if it loses focus.
+  final ValueChanged<bool>? onFocusChange;
+
+  /// Determines the way that drag start behavior is handled.
+  ///
+  /// If set to [DragStartBehavior.start], the drag behavior used to move the
```
Comment (kawaijoe) [inline forui/lib/src/widgets/switch.dart:66]: This feels kind of confusing tbh

File: forui/lib/src/widgets/switch.dart [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+    required this.thumb,
+    required this.focus,
+  });
+
+  /// Creates a [FSwitchStyle] that inherits its properties from [colorScheme].
+  FSwitchStyle.inherit({required FColorScheme colorScheme})
+      : checked = colorScheme.primary,
+        unchecked = colorScheme.border,
+        thumb = colorScheme.background,
+        focus = HSLColor.fromColor(colorScheme.primary.withOpacity(0.80))
```
Comment (kawaijoe) [inline forui/lib/src/widgets/switch.dart:149]: Do we need to convert this to `HSLColor`?

File: forui/lib/src/widgets/switch.dart (hunk 2 above)
Comment (Pante) [reply forui/lib/src/widgets/switch.dart:66]: It was copied from CupertinoSwitch. Out of curiosity, which part is confusing?

File: forui/lib/src/widgets/switch.dart (hunk 1 above)
Comment (Pante) [reply forui/lib/src/widgets/switch.dart:82]: Good catch!

File: forui/lib/src/widgets/switch.dart (hunk 3 above)
Comment (Pante) [reply forui/lib/src/widgets/switch.dart:149]: This code is also copied from CupertinoSwitch. There probably is a way to do it but I'm not too sure.

**Pull request merged**